docs/release_notes/release-3.1.md

```diff
@@ -0,0 +1,119 @@
+
+### Improvements
+
+#### Shared-data cluster
+
+- Optimized the data cache in StarRocks shared-data clusters. The optimized data cache allows for specifying the range of hot data. It can also prevent queries against cold data from occupying the local disk cache, thereby ensuring the performance of queries against hot data.
+
+#### Materialized view
+
+- Optimized the creation of an asynchronous materialized view:
+  - Supports random bucketing. If users do not specify bucketing columns, StarRocks adopts random bucketing by default.
+  - Supports using `ORDER BY` to specify a sort key.
+  - Supports specifying attributes such as `colocate_group`, `storage_medium`, and `storage_cooldown_time`.
+  - Supports using session variables. Users can configure these variables by using the `properties("session.<variable_name>" = "<value>")` syntax to flexibly adjust view refreshing strategies.
+  - Supports creating materialized views based on views. This makes materialized views easier to use in data modeling scenarios, because users can flexibly use views and materialized views based on their varying needs to implement layered modeling.
+- Optimized query rewrite with asynchronous materialized views:
+  - Supports Stale Rewrite, which allows materialized views that are not refreshed within a specified time interval to be used for query rewrite regardless of whether the base tables of the materialized views are updated. Users can specify the time interval by using the `mv_rewrite_staleness_second` property at materialized view creation.
+  - Supports rewriting View Delta Join queries against materialized views that are created on Hive catalog tables (a primary key and a foreign key must be defined).
+  - Optimized the mechanism for rewriting queries that contain union operations, and supports rewriting queries that contain joins or functions such as COUNT DISTINCT and time_slice.
+- Optimized the refreshing of asynchronous materialized views:
+  - Optimized the mechanism for refreshing materialized views that are created on Hive catalog tables. StarRocks now can perceive partition-level data changes, and refreshes only the partitions with data changes during each automatic refresh.
+  - Supports using the `REFRESH MATERIALIZED VIEW WITH SYNC MODE` syntax to synchronously invoke materialized view refresh tasks.
+- Enhanced the use of asynchronous materialized views:
+  - Supports using `ALTER MATERIALIZED VIEW {ACTIVE | INACTIVE}` to enable or disable a materialized view. Materialized views that are disabled (in the `INACTIVE` state) cannot be refreshed or used for query rewrite, but can be directly queried.
+  - Supports using `ALTER MATERIALIZED VIEW SWAP WITH` to replace a materialized view. Users can create a new materialized view and then perform an atomic replacement on an existing materialized view to implement schema changes on the existing materialized view.
```

Comment:
atomic replacement, atomic exchange, atomic swap in release notes and docs --- suggest using consistent terms